client/state-db/src/pruning.rs

```diff
@@ -519,10 +398,10 @@ impl<BlockHash: Hash, Key: Hash, D: MetaDb> RefWindow<BlockHash, Key, D> {
 		number: u64,
 		commit: &mut CommitSet<Key>,
 	) -> Result<(), Error<D::Error>> {
-		if self.base == 0 && self.queue.len() == 0 && number > 0 {
+		if self.base == 0 && self.is_empty() && number > 0 {
 			// assume that parent was canonicalized
 			self.base = number;
-		} else if (self.base + self.queue.len() as u64) != number {
+		} else if (self.base + self.queue.len(self.base) as u64) != number {
```

Comment:
I see "self.base + self.queue.len(self.base) as u64" in two place at least and each time I got to read the queue.len function. Maybe a function for this expression would be better than the current queue.len function  (I am not sure if it can be removed, only window_size really matters).